includes/widget/class-urlslab-widget-cache.php

```diff
@@ -255,63 +261,51 @@ function resource_hints( $hints, $relation_type ) {
 		return $hints;
 	}
 
-
-	private function get_cache_valid_from() {
-		$post_time = get_the_modified_time( 'U' );
-		if ( $post_time ) {
-			return max( self::$active_rule->get_valid_from(), $post_time );
+	public function set_404() {
+		if ( $this->get_option( self::SETTING_NAME_CACHE_404 ) ) {
+			$this->init_check( true );
 		}
+	}
 
-		return self::$active_rule->get_valid_from();
+	private function get_ip_lock_file_name( $ip ): string {
+		return wp_upload_dir()['basedir'] . '/urlslab/' . md5( $ip ) . '_lock.html';
 	}
 
-	public function init_check( $is_404 = false ) {
-		if ( ! $is_404 && ( ! $this->is_cache_enabled() || ! $this->start_rule() ) ) {
-			self::$cache_enabled = false;
 
-			return;
+	public function init_check( $is_404 = false ) {
+		if ( isset( $_SERVER['UL_CV'] ) && is_numeric( $_SERVER['UL_CV'] ) && $_SERVER['UL_CV'] !== $this->get_option( self::SETTING_NAME_CACHE_VALID_FROM ) ) {
```

Comment:
sanitization also here

Reply:
nooooooo :)
Sanitization needs to be done justy in case it has sense .... also suggest single type of attack if you pass through the condition is_numeric()